fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyWritePlatformServiceImpl.java

```diff
@@ -148,7 +148,7 @@ public CommandProcessingResult deleteDelinquencyBucket(Long delinquencyBucketId,
     public CommandProcessingResult applyDelinquencyTagToLoan(Long loanId, JsonCommand command) {
         Map<String, Object> changes = new HashMap<>();
 
-        final Loan loan = this.loanRepository.findOneWithNotFoundDetection(loanId);
```

Comment:
As the loan is not used as a reference, rather doing actions on it, we should use the "this.loanRepository.findOneWithNotFoundDetection"

Using the "getReferenceById" is only useful if we dont need to fetch the loan entirely (fields and associations), but only we need the reference of it. 
Example:
`new LoanTransaction(loan, transactionDate,....)`

Here we dont need to fetch the loan, only the reference which can be used to associate the loan entity with the loan transaction entity.

Reply:
Done!

Reply:
@josehernandezfintecheandomx There is no change. You are still using referenceId